### PR TITLE
Implement Nullable Reference Types on Handler projects

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change ###

Implement Nullable Reference Types on Handler projects.
More information: https://github.com/dotnet/maui/issues/232

### PR Checklist ###

- [x] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

File: src/Platform.Handlers/src/Xamarin.Platform.Handlers/Xamarin.Platform.Handlers.csproj [hunk 1]
```diff
@@ -1,8 +1,10 @@
 ﻿<Project Sdk="MSBuild.Sdk.Extras">
   <PropertyGroup>
-    <TargetFrameworks>$(TargetFrameworks);netstandard2.0;Xamarin.iOS10;MonoAndroid10.0;Xamarin.Mac20</TargetFrameworks>
+    <TargetFrameworks>netstandard2.1;Xamarin.iOS10;MonoAndroid10.0;Xamarin.Mac20</TargetFrameworks>
     <RootNamespace>Xamarin.Platform</RootNamespace>
     <AssemblyName>Xamarin.Platform</AssemblyName>
+    <LangVersion>8.0</LangVersion>
+    <Nullable>enable</Nullable>
```
Comment (jsuarezruiz) [inline src/Platform.Handlers/src/Xamarin.Platform.Handlers/Xamarin.Platform.Handlers.csproj:7]: This is the important change!

**Review by StephaneDelcroix — APPROVED** [review]

File: src/Platform.Handlers/src/Xamarin.Platform.Handlers/Xamarin.Platform.Handlers.csproj (hunk 1 above)
Comment (StephaneDelcroix) [reply src/Platform.Handlers/src/Xamarin.Platform.Handlers/Xamarin.Platform.Handlers.csproj:7]: I'd go one step further, enable it in Directory.Build.props, and disable it on projects (almost all) that aren't ready to support it yet. Same effect, different message

File: src/Platform.Handlers/src/Xamarin.Platform.Handlers/Handlers/Button/ButtonHandler.iOS.cs [hunk 2]
```diff
@@ -9,7 +8,8 @@ public partial class ButtonHandler : AbstractViewHandler<IButton, UIButton>
 
 		public static void MapText(IViewHandler handler, IButton view)
 		{
-			((UIButton)handler.NativeView).SetText(view.Text);
+			if (handler.NativeView is UIButton uIButton)
+				uIButton.SetText(view.Text);
```
Comment (StephaneDelcroix) [inline src/Platform.Handlers/src/Xamarin.Platform.Handlers/Handlers/Button/ButtonHandler.iOS.cs:12]: checking that NativeView isn't null isn't enough, we probably should check wether it's disposed. Or are setting NativeView to null when we dispose it ?

also, why do we need casting ? don't we have a generic type parameter of type UIButton ?

File: src/Platform.Handlers/src/Xamarin.Platform.Handlers/Handlers/IViewHandler.Android.cs [hunk 3]
```diff
@@ -7,6 +7,6 @@ public interface IAndroidViewHandler : IViewHandler
 	{
 		void SetContext(Context context);
 
-		AView View { get; }
+		AView? View { get; }
```
Comment (StephaneDelcroix) [inline src/Platform.Handlers/src/Xamarin.Platform.Handlers/Handlers/IViewHandler.Android.cs:10]: everything would be so simple, or at least simpler, if the View was never null. could we consider this ? (View created as part of the ctor, e.g.)

**Review by StephaneDelcroix — CHANGES_REQUESTED** [review]
making views nullable adds more check than we had, so maybe the view doesn't need to be nullable after all ?

File: src/Platform.Handlers/src/Xamarin.Platform.Handlers/Xamarin.Platform.Handlers.csproj (hunk 1 above)
Comment (jsuarezruiz) [reply src/Platform.Handlers/src/Xamarin.Platform.Handlers/Xamarin.Platform.Handlers.csproj:6]: Nice suggestion but currently can be only enable in the handlers project and 8.0 is already all setup everywhere.

File: src/Platform.Handlers/src/Xamarin.Platform.Handlers/Handlers/IViewHandler.Android.cs (hunk 3 above)
Comment (PureWeen) [reply src/Platform.Handlers/src/Xamarin.Platform.Handlers/Handlers/IViewHandler.Android.cs:10]: Agreed

What in here that's marked nullable can we guarantee won't be null.

Like does a ViewHandler with a null View make any sense?

**Review by PureWeen — CHANGES_REQUESTED** [review]

File: src/Platform.Handlers/src/Xamarin.Platform.Handlers/Handlers/IViewHandler.Android.cs (hunk 3 above)
Comment (PureWeen) [reply src/Platform.Handlers/src/Xamarin.Platform.Handlers/Handlers/IViewHandler.Android.cs:10]: The views are created after the ctror and can be swapped out

https://github.com/xamarin/Xamarin.Forms/blob/7aedbc7d3e592a531879d2eb05a6daa673e16715/src/Platform.Handlers/src/Xamarin.Platform.Handlers/Handlers/View/AbstractViewHandler.cs#L43

I'm assuming we'll need this to be nullable

File: src/Platform.Handlers/src/Xamarin.Platform.Handlers/Handlers/View/AbstractViewHandler.cs [hunk 4]
```diff
@@ -20,25 +18,25 @@ public abstract partial class AbstractViewHandler<TVirtualView, TNativeView> : I
...
 		protected AbstractViewHandler(PropertyMapper mapper)
 		{
-			defaultMapper = mapper;
+			_defaultMapper = mapper;
 		}
 
 		protected abstract TNativeView CreateView();
 
-		public NativeView View => TypedNativeView;
+		protected TNativeView TypedNativeView { get; private set; } = default!;
```
Comment (PureWeen) [inline src/Platform.Handlers/src/Xamarin.Platform.Handlers/Handlers/View/AbstractViewHandler.cs:33]: Why did you change the access modifiers on these?

**Review by PureWeen — APPROVED** [review]
